### PR TITLE
[Docs] Japanese translation of docs/internals_gpio_control.md

### PR DESCRIPTION
**Pull request by umi-umi**
## Description

Add a Japanese translation of docs/internals_gpio_control.md.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [x] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by mtei** [comment]
@shelaf @s-show @hsgw @satt99 @takashiski @yoshihara よかったらレビューお願いします。

File: docs/ja/internals_gpio_control.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+| `setPinInputHigh(pin)` | ピンを組み込みの組み込みのプルアップ抵抗の入力として設定 | `DDRB &= ~(1<<2); PORTB \|= (1<<2)` | `palSetLineMode(pin, PAL_MODE_INPUT_PULLUP)` |
+| `setPinInputLow(pin)` | ピンを組み込みのプルダウン抵抗の入力として設定 | N/A (AVR ではサポートされません) | `palSetLineMode(pin, PAL_MODE_INPUT_PULLDOWN)` |
+| `setPinOutput(pin)` | ピンを出力として設定 | `DDRB \|= (1<<2)` | `palSetLineMode(pin, PAL_MODE_OUTPUT_PUSHPULL)` |
+| `writePinHigh(pin)` | ピンを出力と仮定して、ピンレベルを high に設定 | `PORTB \|= (1<<2)` | `palSetLine(pin)` |
+| `writePinLow(pin)` | ピンを出力と仮定して、ピンレベルを low に設定 | `PORTB &= ~(1<<2)` | `palClearLine(pin)` |
+| `writePin(pin, level)` | ピンを出力と仮定して、ピンレベルを設定 | `(level) ? PORTB \|= (1<<2) : PORTB &= ~(1<<2)` | `(level) ? palSetLine(pin) : palClearLine(pin)` |
+| `readPin(pin)` | ピンのレベルを返す | `_SFR_IO8(pin >> 4) & _BV(pin & 0xF)` | `palReadLine(pin)` |
+| `togglePin(pin)` | ピンを出力と仮定して、ピンレベルを反転 | `PORTB ^= (1<<2)` | `palToggleLine(pin)` |
+
+## 追加の設定 :id=advanced-settings
```
Comment (s-show) [inline docs/ja/internals_gpio_control.md:26]: 「追加の設定」となる英語は「additional settings」ではないかと思います。
「advanced-settings」は「高度な設定」と訳すべきだと思いますし、本文では「高度な設定」と訳していますので、見出しも同じにした方が良いと思います。
```suggestion
## 高度な設定 :id=advanced-settings
```

Comment (umi-umi) [reply docs/ja/internals_gpio_control.md]: ご指摘ありがとうございます。

**Comment by shelaf** [comment]
internalsで始まるドキュメントは翻訳の優先度は低いと思うので、できれば他のからお願いします🙏

File: docs/ja/internals_gpio_control.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+QMK には、マイクロコントローラに依存しない GPIO 制御抽象レイヤーがあります。これは異なるプラットフォーム間でピン制御に簡単にアクセスできるようにするためのものです。
+
+## 関数 :id=functions
+
+以下の関数は GPIO の基本的な制御を提供し、`quantum/quantum.h` にあります。
+
+| 関数 | 説明 | 古い AVR の例 | 古い ChibiOS/ARM の例 |
+|------------------------|--------------------------------------------------|-------------------------------------------------|-------------------------------------------------|
+| `setPinInput(pin)` | ピンを高インピーダンス(High-Z)の入力として設定 | `DDRB &= ~(1<<2)` | `palSetLineMode(pin, PAL_MODE_INPUT)` |
+| `setPinInputHigh(pin)` | ピンを組み込みのプルアップ抵抗の入力として設定 | `DDRB &= ~(1<<2); PORTB \|= (1<<2)` | `palSetLineMode(pin, PAL_MODE_INPUT_PULLUP)` |
```
Comment (mtei) [inline docs/ja/internals_gpio_control.md:17]:
```suggestion
| `setPinInputHigh(pin)` | ピンを組み込みのプルアップ抵抗付きの入力として設定 | `DDRB &= ~(1<<2); PORTB \|= (1<<2)` | `palSetLineMode(pin, PAL_MODE_INPUT_PULLUP)` |
```

File: docs/ja/internals_gpio_control.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+| `setPinInputLow(pin)` | ピンを組み込みのプルダウン抵抗の入力として設定 | N/A (AVR ではサポートされません) | `palSetLineMode(pin, PAL_MODE_INPUT_PULLDOWN)` |
```
Comment (mtei) [inline docs/ja/internals_gpio_control.md:18]:
```suggestion
| `setPinInputLow(pin)` | ピンを組み込みのプルダウン抵抗付きの入力として設定 | N/A (AVR ではサポートされません) | `palSetLineMode(pin, PAL_MODE_INPUT_PULLDOWN)` |
```

File: docs/ja/internals_gpio_control.md [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+| `setPinOutput(pin)` | ピンを出力として設定 | `DDRB \|= (1<<2)` | `palSetLineMode(pin, PAL_MODE_OUTPUT_PUSHPULL)` |
+| `writePinHigh(pin)` | ピンを出力と仮定して、ピンレベルを high に設定 | `PORTB \|= (1<<2)` | `palSetLine(pin)` |
```
Comment (mtei) [inline docs/ja/internals_gpio_control.md:20]:
```suggestion
| `writePinHigh(pin)` | ピンレベルを high に設定 (ピンを出力として設定してあると仮定) | `PORTB \|= (1<<2)` | `palSetLine(pin)` |
```

File: docs/ja/internals_gpio_control.md [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+| `writePinLow(pin)` | ピンを出力と仮定して、ピンレベルを low に設定 | `PORTB &= ~(1<<2)` | `palClearLine(pin)` |
```
Comment (mtei) [inline docs/ja/internals_gpio_control.md:21]:
```suggestion
| `writePinLow(pin)` | ピンレベルを low に設定 (ピンを出力として設定してあると仮定)  | `PORTB &= ~(1<<2)` | `palClearLine(pin)` |
```

File: docs/ja/internals_gpio_control.md [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+| `writePin(pin, level)` | ピンを出力と仮定して、ピンレベルを設定 | `(level) ? PORTB \|= (1<<2) : PORTB &= ~(1<<2)` | `(level) ? palSetLine(pin) : palClearLine(pin)` |
```
Comment (mtei) [inline docs/ja/internals_gpio_control.md:22]:
```suggestion
| `writePin(pin, level)` | ピンレベルを設定 (ピンを出力として設定してあると仮定)  | `(level) ? PORTB \|= (1<<2) : PORTB &= ~(1<<2)` | `(level) ? palSetLine(pin) : palClearLine(pin)` |
```

File: docs/ja/internals_gpio_control.md [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+| `readPin(pin)` | ピンのレベルを返す | `_SFR_IO8(pin >> 4) & _BV(pin & 0xF)` | `palReadLine(pin)` |
+| `togglePin(pin)` | ピンを出力と仮定して、ピンレベルを反転 | `PORTB ^= (1<<2)` | `palToggleLine(pin)` |
```
Comment (mtei) [inline docs/ja/internals_gpio_control.md:24]:
```suggestion
| `togglePin(pin)` | ピンレベルを反転 (ピンを出力として設定してあると仮定)  | `PORTB ^= (1<<2)` | `palToggleLine(pin)` |
```

File: docs/ja/internals_gpio_control.md (hunk 2 above)
Comment (umi-umi) [reply docs/ja/internals_gpio_control.md]: ご指摘ありがとうございます。

**Comment by mtei** [comment]
I think this is ready to merge.

**Comment by noroadsleft** [comment]
ありがとうございました！

**Pull request merged**